### PR TITLE
Container image for kfn

### LINKED ISSUE #3: Create and publish a docker image
The CLI tool should be able to run as a container so that it can easily be integrated into a Tekton pipeline.

### PR DESCRIPTION
**Pull request by slinkydeveloper**
Closes #3

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,20 @@
+FROM fedora:latest
```
Comment (lance) [inline Dockerfile]: Why not from `registry.access.redhat.com/ubi7/ubi:latest`?

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+# Don't include container-selinux and remove
+# directories used by dnf that are just taking
+# up space.
+RUN yum -y install buildah fuse-overlayfs --exclude container-selinux; rm -rf /var/cache /var/log/dnf* /var/log/yum.*
+
+# Adjust storage.conf to enable Fuse storage.
+RUN sed -i -e 's|^#mount_program|mount_program|g' -e '/additionalimage.*/a "/var/lib/shared",' /etc/containers/storage.conf
+RUN mkdir -p /var/lib/shared/overlay-images /var/lib/shared/overlay-layers; touch /var/lib/shared/overlay-images/images.lock; touch /var/lib/shared/overlay-layers/layers.lock
```
Comment (lance) [inline Dockerfile]: I recommend combining these three lines into a single `RUN` command to minimize the number of layers in the image.

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+COPY kfn /usr/bin
+
+# Set up environment variables to note that this is
+# not starting with usernamespace and default to
+# isolate the filesystem with chroot.
+ENV BUILDAH_ISOLATION=chroot
+ENV KFN_IN_CLUSTER=true
+
+ENTRYPOINT ["kfn"]
```
Comment (lance) [inline Dockerfile]: Nit: add newline at end of file

File: Makefile [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+check:
+	ifeq (, $(shell which packr))
+	$(error "No packr in $(PATH), consider installing https://github.com/gobuffalo/packr")
+	endif
+
+build:
+	packr build -v -o kfn main.go
+
+image:
+	buildah bud --tag kfn .
```
Comment (lance) [inline Makefile]: Nit: newline at end of file

File: Dockerfile (hunk 1 above)
Comment (slinkydeveloper) [reply Dockerfile]: I took the docker image example from buildah repo. tbh I don't know what's the difference here, can you explain it?

Comment (slinkydeveloper) [reply Dockerfile]: why not `ubi-minimal`?

Comment (lance) [reply Dockerfile]: It could very well be ubi-minimal, depending on the needs of the `kfn` but I imagine that would be fine.

Comment (slinkydeveloper) [reply Dockerfile]: Both ubi and ubi-minimal doesn't work due missing packages. I prefer to leave fedora for now, we'll improve it in the future

Comment (slinkydeveloper) [reply Dockerfile]: Since we need this image now only to test the integration with the operator.

**Pull request merged**